pkg/cmd/pulumi/state.go

```diff
@@ -193,3 +195,70 @@ func totalStateEdit(ctx context.Context, s backend.Stack, showPrompt bool, opts
 	}
 	return result.WrapIfNonNil(s.ImportDeployment(ctx, &dep))
 }
+
+// Prompt the user to select a URN from the passed in state.
+//
+// stackName is the name of the current stack.
+//
+// snap is the snapshot of the current stack. snap cannot be nil, but (*snap) may be nil.
+// (*snap) will be set to the retrieved snapshot value is it is nil.
+//
+// Prompt is displayed to the user when selecting the URN.
+func getURNFromState(
+	ctx context.Context, stackName string, snap **deploy.Snapshot, prompt string,
+) (resource.URN, error) {
+	if snap == nil {
```

Comment:
doc comment says "snap cannot be nil", but then your allowing it to be nil here. Suspect docs should be changed.